pdbufr/bufr_structure.py

```diff
@@ -174,6 +246,8 @@ def extract_observations(
         subset_count = message["numberOfSubsets"]
     else:
         subset_count = 1
+        
+    filters_without_computed = {k:v for k,v in filters.items() if not k in computed_keys}        
```

Comment:
It is possible to move this line out of `extract_observations` and not change the function? Looks much cleaner to me (but I may miss something).

Reply:
Yes, also a good hint - included